server/resources/resourcesService.js

```diff
@@ -23,12 +23,16 @@ export async function getAll({ draft = false }, { page = 1, perPage = 20 }) {
 		limit: perPage,
 		offset: (page - 1) * perPage,
 	});
+
+	const { pagedResult, findAllResource } = resources;
+
 	const totalCount = await repository.count({ draft });
 	return {
 		lastPage: Math.ceil(totalCount / perPage) || 1,
 		page,
 		perPage,
-		resources,
+		resources: pagedResult,
+		allResources: findAllResource,
```

Comment:
> This makes no sense at all to me. The point of paginating the resources is to allow the site to remain performant as the amount of data grows. If every "page" of resources also actually includes _all of the resources_ as well, there was no point at all in paginating to being with. In fact it's _worse_ than never having paginated, because as many as `perPage` resources are included in the response _twice_.
> 
> Also note there's no point making yet another DB query to get the total count (line 29) when that's the length of the array of all resources (as you actually assert on in the integration test).

you are right  I've got you point we need to handle filtering in backend  using topic id then send send filtered data into front end :+1: